druid/examples/async_event.rs

```diff
@@ -0,0 +1,133 @@
+}
+
+fn generate_colours(event_sink: druid::ExtEventSink) {
+    // This function is called in a separate thread, and runs until the program ends.
+    // We take an `ExtEventSink` as an argument, we can use this event sink to send
+    // commands to the main thread. Every time we generate a new colour we send it
+    // to the main thread.
+    let start_time = Instant::now();
+    let mut color = Color::WHITE;
+
+    loop {
+        let time_since_start = (Instant::now() - start_time).as_nanos();
+        let (r, g, b, _) = color.as_rgba8();
+
+        // there is no logic here; it's a very silly way of mutating the color.
+        color = match (time_since_start % 2, time_since_start % 3) {
+            (0, _) => Color::rgb8(r.wrapping_add(3), g, b),
+            (_, 0) => Color::rgb8(r, g.wrapping_add(3), b),
+            (_, _) => Color::rgb8(r, g, b.wrapping_add(3)),
+        };
+
+        // We send a command to the event_sink. This command will be
+        // send to the widgets, and widgets or controllers can look for this
+        // event. We give it the data associated with the event and a target.
+        // In this case this is just `Target::Auto`, look at the identity example
+        // for more detail on how to send commands to specific widgets.
+        if event_sink
+            .submit_command(SET_COLOR, color.clone(), Target::Auto)
+            .is_err()
+        {
+            break;
+        }
+        thread::sleep(Duration::from_millis(20));
+    }
+}
+
+/// A widget that displays a color.
+struct ColorWell;
+
+impl Widget<Color> for ColorWell {
+    fn event(&mut self, _ctx: &mut EventCtx, event: &Event, data: &mut Color, _env: &Env) {
+        match event {
+            // This is where we handle our command.
+            Event::Command(cmd) if cmd.is(SET_COLOR) => {
+                // We don't do much data processing in the `event` method.
+                // All we really do is just set the data. This causes a call
+                // to `update` which requests a paint. You can also request a paint
+                // during the event, but this should be reserved for changes to self.
+                // For changes to `Data` always make `update` do the paint requesting.
+                *data = cmd.get_unchecked(SET_COLOR).clone();
+            }
+            _ => (),
+        }
+    }
+
+    fn lifecycle(&mut self, _ctx: &mut LifeCycleCtx, _event: &LifeCycle, _data: &Color, _: &Env) {}
+
+    fn update(&mut self, ctx: &mut UpdateCtx, old_data: &Color, data: &Color, _: &Env) {
+        if old_data != data {
+            ctx.request_paint()
+        }
+    }
+
+    fn layout(&mut self, _: &mut LayoutCtx, bc: &BoxConstraints, _: &Color, _: &Env) -> Size {
+        bc.max()
+    }
+
+    fn paint(&mut self, ctx: &mut PaintCtx, data: &Color, _env: &Env) {
+        let rect = ctx.size().to_rounded_rect(5.0);
+        ctx.fill(rect, data);
+    }
+}
+
+fn make_ui() -> impl Widget<Color> {
+    ColorWell {}
```

Comment:
I don't think these braces are necessary?

```suggestion
    ColorWell
```